__tests__/e2e/scripts/test.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+# variables
+KIBANA_PORT=5601
+
+# ensure Docker is running
+docker ps &> /dev/null
+if [ $? -ne 0 ]; then
+    echo "⚠️  Please start Docker"
+    exit 1
+fi
+
+# formatting
+bold=$(tput bold)
+normal=$(tput sgr0)
+
+# Wait for Kibana to start
+##################################################
+echo "" # newline
+echo "${bold}Waiting for Kibana to start...${normal}"
+yarn wait-on -i 500 -w 500 http-get://admin:changeme@localhost:$KIBANA_PORT/api/status > /dev/null
```

Comment:
@jahtalab is there a way we can improve this command?

Reply:
I think we should refactor the setup into the docker-compose, so we can add dependency for both ES and kibana and docker-compose would make sure to bring up those before running the tests.

Reply:
dependency won't make sure kibana is up, it only means kibana docker container is up. Kibana itself takes a while to start.

Reply:
Yes, that's right, however, I do think it makes more sense to just use a docker compose file for the setup of synthetics, and for that matter, we should probably just get rid of run.sh eventually and replace it with docker-compose.

We can monitor for the synthetics container starting by checking the docker logs.

I'm not sure if the shell scripts from APM, in the end, are that great. I can see things being much simpler without them, we just need to:

1. Run `docker-compose up`
2. Wait for kibana to start
3. Run the synthetics tests in a loop for some length of time until they pass

We could optionally have a step between 2 and 3 where we monitor the output of heartbeat waiting for the tests to run, but I don't think we really need that. We're testing the functionality end to end after all, we just care that results eventually appear in Uptime

Reply:
@shahzad31 , we can add a healthcheck to make sure kibana is actually up before running the tests.

Reply:
@jahtalab i have added health checks